eth/filters/statediff.go

```diff
@@ -0,0 +1,114 @@
+package filters
+
+import (
+	"math/big"
+	"reflect"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+// ProcessStateChanges builds the state diff Payload from the modified accounts in the StateChangeEvent
+var emptyPayload Payload
+
+func ProcessStateChanges(event core.StateChangeEvent, crit ethereum.FilterQuery) (Payload, error) {
+	var accountDiffs []AccountDiff
+	block := event.Block
+	// Iterate over state changes to build AccountDiffs
+	for addr, modifiedAccount := range event.StateChanges {
+		if len(crit.Addresses) > 0 && !includes(crit.Addresses, addr){
+			continue
+		}
+
+		a, err := buildAccountDiff(addr, modifiedAccount)
+		if err != nil {
+			return emptyPayload, err
+		}
+
+		accountDiffs = append(accountDiffs, a)
+	}
+
+	if len(accountDiffs) == 0 {
+		return emptyPayload, nil
+	}
+
+	stateDiff := StateDiff{
+		BlockNumber:     block.Number(),
+		BlockHash:       block.Hash(),
+		UpdatedAccounts: accountDiffs,
+	}
+
+	stateDiffRlp, err := rlp.EncodeToBytes(stateDiff)
```

Comment:
I'm wondering if encoding this to an RLP makes sense still. 🤔 Thinking of leaving it as is for now, but curious what folks think.